### ISSUE TITLE
Unable to read package from path 'chocolatey\chocolatey.nupkg'

### ISSUE THREAD

**Issue by BiddlyWiddly** [chocolatey/home#100]
Hope this is the place to ask for help? If not, please point me in the right direction.

Been using Chocolatey via Chocolatey GUI for some time, with no issues.

Laptop battery died when updating Chocolatey itself via the GUI (didn't realise battery had run down while in sleep mode overnight - facepalm!) and upon reboot, the installation seemed to have corrupted and was failing with reference to chocolatey.config

Have managed to get past that (removed file from folder and it created a new version) but now everything I try to install either using GUI or CLI fails with the above message: Unable to read package from path 'chocolatey\chocolatey.nupkg'

Any ideas?

**Comment by gep13** [comment]
@BiddlyWiddly sorry to hear that you are having issues!

If you navigate to the following path:

```
C:\ProgramData\chocolatey\lib\chocolatey
```

Can you confirm what is in that folder?

**Comment by BiddlyWiddly** [comment]
Thanks, and no apologies necessary - not your fault, more down to my rookie error!

Contents of that path shown below.

![image](https://user-images.githubusercontent.com/4438718/150795240-3cdd1576-bf4d-48db-b84f-b97f194b3bd3.png)

**Comment by gep13** [comment]
At this point, I would be concerned that this nupkg has been corrupted, so I would suggest downloading it again:

v0.12.0
https://community.chocolatey.org/api/v2/package/chocolatey/0.12.0

v0.11.3
https://community.chocolatey.org/api/v2/package/chocolatey/0.11.3

And then replacing the nupkg in that folder, then try running things again.

If that doesn't work, we can move to a more involved process.

**Comment by BiddlyWiddly** [comment]
Thanks for that - have downloaded 0.12.0 (the version I'm running) and it's moved us past that error, but now an issue with chocolateyInstall.ps1. Copied from Powershell window:

_ERROR: Exception calling "Start" with "0" argument(s): "The application has failed to start because its side-by-side configuration is incorrect. Please see the application event log or use the command-line sxstrace.exe tool for more detail"
The upgrade of ccleaner was NOT successful.
Error while running 'C:\ProgramData\chocolatey\lib\ccleaner\tools\chocolateyInstall.ps1'.
 See log for details._

**Comment by gep13** [comment]
I am not sure if I follow exactly what you have done here.

Can you provide an exact set of steps that you have followed between downloading/copying the nupkg, to getting the error that you are now seeing?

It would also be helpful if you could include the contents of the chocolatey.log file for the operation(s) that you are attempting.

**Comment by BiddlyWiddly** [comment]
Sure, no probs.

Downloaded the v0.12.0 file from the link you provided - filename was chocolatey.0.12.0.nupkg.

Copied the downloaded file into `C:\ProgramData\chocolatey\lib\chocolatey` and removed the existing chocolatey.nupkg (cut and pasted it onto my desktop for posterity).

Folder contents of `C:\ProgramData\chocolatey\lib\chocolatey` as follows:

![image](https://user-images.githubusercontent.com/4438718/150960286-8b6f4d2d-a2d8-43fe-9002-b360873c2462.png)

From there, I opened Chocolatey GUI to see what programs/apps needed updating, Ccleaner being one of them. 

So I could easily copy out any error text, I opened a Powershell (admin) window, and ran command `choco upgrade ccleaner`

Powershell window in its entirety:

![image](https://user-images.githubusercontent.com/4438718/150960789-fc8c5bd3-0ca0-44b7-86ef-330d63ef8bb4.png)

Have copied out the relevant part of chocolatey.log into the attached file logfile.txt
[logfile.txt](https://github.com/chocolatey/home/files/7932786/logfile.txt)

**Comment by gep13** [comment]
@BiddlyWiddly this doesn't seem like a problem with Chocolatey, but rather with the package itself.

I have just attempted the installation of this package, and it worked without any problems:

![image](https://user-images.githubusercontent.com/1271146/151322458-a7364805-c9b8-49b9-bd03-ac5f3dc415b8.png)

Perhaps there is an issue with upgrading the application if it is already installed, but that would be best brought up with the package maintainer, not here on the Chocolatey repository.

**Comment by BiddlyWiddly** [comment]
It's so weird - was getting that error regardless of what package I tried to update (there were about four different programs/apps to update).

Came to it this morning, saw your comment, so thought I'd see if I could run a `choco upgrade` command for Chocolatey itself. Lo and behold, it worked! It found an update to go from v0.12.0 to v0.12.1 which installed successfully.

Went back and tried `choco upgrade ccleaner` and that worked too, no issues.

Ran `choco upgrade all`, identified 7 packages that needed updating - said 'y' to four and 'n' to three so that I could leave something to try in Chocolatey GUI. All four updated successfully through the Powershell command

Jumped back over to Chocolatey GUI to see if the remaining ones would update (I usually use the GUI as I'm not an experienced CLI user) and they updated fine through GUI too.

tl;dr: Managed to upgrade Chocolatey itself, now other packages will update through CLI and GUI.

Thanks for your help!!

**Comment by gep13** [comment]
@BiddlyWiddly I am very glad to hear that this is working again for you. Sounds like there might have been _something_ not quite aligned correctly after the power outage, and the upgrade has sorted things out.  I will go ahead and close this out.
